pkg/backends/calico/client.go

```diff
@@ -421,14 +440,34 @@ func (c *client) updatePeersV1() {
 					peers = append(peers, c.nodeAsBGPPeers(peerNodeName)...)
 				}
 			} else {
-				ip := cnet.ParseIP(v3res.Spec.PeerIP)
+				// Separate port from Ip if it uses <ip>:<port> format
+				host, port := parseIPPort(v3res.Spec.PeerIP)
+				ip := cnet.ParseIP(host)
 				if ip == nil {
 					log.Warning("PeerIP is not assigned or is malformed")
 					continue
 				}
+
+				// If port is not empty, we use the given value to peer.
+				// If port is empty, check if BGP Peer is a calico/node, and use its listenPort to peer.
+				if port == "" {
+					// If port is empty, nodesWithIPPortAndAS() returns list of calico/node that matches IP and ASNumber.
+					nodeNames := c.nodesWithIPPortAndAS(host, v3res.Spec.ASNumber, port)
+					if len(nodeNames) != 0 {
+						listenPort := ""
```

Comment:
nit: please delete this, and assign directly to `port`, instead of to `listenPort` and then to `port`.  (Just because I think the addition of `listenPort` makes the code here look more complex than it really is.  But please do correct me if I'm missing something.)